src/nrc/api/tests/test_scopes.py

```diff
@@ -0,0 +1,361 @@
+    ZDS_CLIENT_CLASS="vng_api_common.mocks.MockClient",
+)
+class ScopeSubscriptionsRead(JWTAuthMixin, APITestCase):
+    scopes = [SCOPE_SUBSCRIPTIONS_READ]
+    source = "urn:nld:oin:00000001234567890000:systeem:Zaaksysteem"
+    sink = "https://endpoint.example.com/webhook"
+    data = {
+        "protocol": ProtocolChoices.HTTP,
+        "source": source,
+        "sink": sink,
+        "types": [
+            "Type A",
+            "Type B",
+        ],
+    }
+
+    def test_correct_scope(self):
+        """
+        test /subscriptions GET:
+        retrieve subscription with correct scope SCOPE_SUBSCRIPTIONS_READ
+        """
+        subscription = SubscriptionFactory.create()
+
+        subscription_url = get_operation_url(
+            "subscription_update", uuid=subscription.uuid
+        )
+
+        with requests_mock.mock() as m:
+            m.register_uri("POST", self.sink, status_code=204)
+            response = self.client.get(subscription_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+    def test_incorrect_scope(self):
+        """
+        test /subscriptions POST:
+        create subscription with incorrect scope SCOPE_SUBSCRIPTIONS_READ
+        """
+        subscription_create_url = get_operation_url("subscription_create")
+
+        with requests_mock.mock() as m:
+            m.register_uri(
+                "POST",
+                self.sink,
+                status_code=204,
+            )
+            response = self.client.post(subscription_create_url, self.data)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
+
+@override_settings(
+    LINK_FETCHER="vng_api_common.mocks.link_fetcher_200",
+    ZDS_CLIENT_CLASS="vng_api_common.mocks.MockClient",
+)
+class ScopeDomainsRead(JWTAuthMixin, APITestCase):
+    scopes = [SCOPE_DOMAINS_READ]
+
+    def test_correct_scope(self):
+        """
+        test /domains GET:
+        retrieve domain with correct scope SCOPE_DOMAINS_READ
+        """
+        domain = DomainFactory()
+
+        url = get_operation_url("domain_read", uuid=domain.uuid)
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_incorrect_scope(self):
+        """
+        test /domains POST:
+        retrieve domain with incorrect scope SCOPE_DOMAINS_READ
+        """
+        data = {"name": "zaken", "documentation_link": "https://example.com/doc"}
+
+        domain_create_url = get_operation_url("domain_create")
+        response = self.client.post(domain_create_url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
+
+@patch("nrc.api.serializers.deliver_message.delay")
+@override_settings(
+    LINK_FETCHER="vng_api_common.mocks.link_fetcher_200",
+    ZDS_CLIENT_CLASS="vng_api_common.mocks.MockClient",
+)
+class ScopeEventsCreate(JWTAuthMixin, APITestCase):
```

Comment:
In de `TestCase` voor `Events` zou ik ook nog een test verwachten die het gedrag test wanneer je niet de scope hebt.